packages/axelar-wasm-std/src/nonempty/hexbinary.rs

```diff
@@ -0,0 +1,68 @@
+use std::ops::Deref;
+
+use cosmwasm_schema::cw_serde;
+
+use crate::nonempty::Error;
+
+#[cw_serde]
+#[serde(try_from = "cosmwasm_std::HexBinary")]
+#[derive(Eq, Hash)]
+pub struct HexBinary(cosmwasm_std::HexBinary);
+
+impl TryFrom<cosmwasm_std::HexBinary> for HexBinary {
+    type Error = Error;
+
+    fn try_from(value: cosmwasm_std::HexBinary) -> Result<Self, Self::Error> {
+        if value.is_empty() {
+            Err(Error::InvalidValue("empty".to_string()))
+        } else {
+            Ok(HexBinary(value))
+        }
+    }
+}
+
+impl TryFrom<std::vec::Vec<u8>> for HexBinary {
+    type Error = Error;
+
+    fn try_from(value: std::vec::Vec<u8>) -> Result<Self, Self::Error> {
+        cosmwasm_std::HexBinary::from(value).try_into()
+    }
+}
+
+impl From<HexBinary> for cosmwasm_std::HexBinary {
+    fn from(value: HexBinary) -> Self {
+        value.0
+    }
+}
+
+impl From<HexBinary> for std::vec::Vec<u8> {
+    fn from(value: HexBinary) -> Self {
+        value.0.into()
+    }
+}
+
+impl Deref for HexBinary {
+    type Target = cosmwasm_std::HexBinary;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::nonempty::{Error, HexBinary};
+
+    #[test]
+    fn test_non_empty_hexbinary() {
+        assert!(HexBinary::try_from(cosmwasm_std::HexBinary::from(&[1, 2, 3])).is_ok())
+    }
+
+    #[test]
+    fn test_empty_non_empty_hexbinary() {
+        assert_eq!(
+            HexBinary::try_from(cosmwasm_std::HexBinary::from(&[])).unwrap_err(),
+            Error::InvalidValue("empty".to_string())
```

Comment:
does it work with err_contains?

Reply:
`Error` doesn't implement `IntoContractError` trait, and it cannot implement it since the macro is defined within the same crate (since it can't refer itself `axelar_wasm_std` when expanded). I don't know of a solution where Rust handles local vs foreign namespacing in the macro